pkg/apiserver/registry/security/workloadidentity/storage/jwt_test.go

```diff
@@ -0,0 +1,613 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package storage
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"time"
+
+	"github.com/go-jose/go-jose/v4/jwt"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/keyutil"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	securityapi "github.com/gardener/gardener/pkg/apis/security"
+	gardencoreinformers "github.com/gardener/gardener/pkg/client/core/informers/externalversions"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+)
+
+var (
+	rsaPrivateKey   *rsa.PrivateKey
+	ecdsaPrivateKey *ecdsa.PrivateKey
+)
+
+var _ = BeforeSuite(func() {
+	rsaKey, err := rsa.GenerateKey(rand.Reader, 4096)
+	Expect(err).ToNot(HaveOccurred())
+	rsaPrivateKey = rsaKey
+
+	ecdsaKey, err := ecdsa.GenerateKey(elliptic.P521(), rand.Reader)
+	Expect(err).ToNot(HaveOccurred())
+	ecdsaPrivateKey = ecdsaKey
+})
+
+var _ = Describe("#JWT", func() {
+	Context("#getClaims", func() {
+		var (
+			issuer          = "https://test.local.gardener.cloud"
+			clusterIdentity = "test-local-gardener-cloud"
+			minDuration     = time.Minute * 10
+			maxDuration     = time.Hour * 48
+			r               = TokenRequestREST{
+				issuer:          issuer,
+				clusterIdentity: clusterIdentity,
+				minDuration:     int64(minDuration.Seconds()),
+				maxDuration:     int64(maxDuration.Seconds()),
+			}
+			workloadName      = "identity"
+			workloadNamespace = "garden-local"
+			workloadUID       = "ab920696-dd12-4723-9bc1-204cfe9edd40"
+			sub               = "gardener.cloud:workloadidentity:" + workloadNamespace + ":" + workloadName + ":" + workloadUID
+			aud               = "gardener.cloud"
+
+			workloadIdentity = &securityapi.WorkloadIdentity{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      workloadName,
+					Namespace: workloadNamespace,
+					UID:       types.UID(workloadUID),
+				},
+				Spec: securityapi.WorkloadIdentitySpec{
+					Audiences: []string{aud},
+				},
+				Status: securityapi.WorkloadIdentityStatus{
+					Sub: sub,
+				},
+			}
```

Comment:
Yes, exactly. When you declare the variables like this, a test could change the variable, so that the subsequent tests then use the changed value. The sequence of tests is not deterministic, so the results would be unpredictable in this case.
You don't change the variables in the tests, but it is a best practice (see [the ginkgo docs](https://onsi.github.io/ginkgo/#extracting-common-setup-beforeeach)) to avoid this when tests are updated in the future. 
Changing the variables to constants works too.